### PR TITLE
JSUI-2367 Accessibility details

### PR DESCRIPTION
**Pull request by ThibodeauJF**
I have requested answers on this JIRA but I guess we can start reviewing this maintenance case
https://coveord.atlassian.net/browse/JSUI-2367
[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/CategoryFacet/CategoryFacet.ts [hunk 1]
```diff
@@ -713,7 +713,11 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
   private buildMoreButton() {
     const svgContainer = $$('span', { className: 'coveo-facet-more-icon' }, SVGIcons.icons.arrowDown).el;
     SVGDom.addClassToSVGInContainer(svgContainer, 'coveo-facet-more-icon-svg');
-    const more = $$('div', { className: 'coveo-category-facet-more', tabindex: 0 }, svgContainer);
+    const more = $$(
```
Comment (samisayegh) [inline src/ui/CategoryFacet/CategoryFacet.ts]: Would it be more consistent to use `new AccessibleButton()` here? Similarly for the `less` button.

File: src/ui/ResponsiveComponents/ResponsiveDropdown/ResponsiveDropdown.ts [hunk 2]
```diff
@@ -80,6 +84,24 @@ export class ResponsiveDropdown {
       .build();
   }
 
+  private closeIfOpened = () => {
+    this.isOpened && this.close();
```
Comment (samisayegh) [inline src/ui/ResponsiveComponents/ResponsiveDropdown/ResponsiveDropdown.ts]: Is it necessary to check if `isOpened` is true, or could we just call `this.close` regardless and remove this method?

File: unitTests/ui/ResponsiveComponents/ResponsiveDropdownTest.ts [hunk 3]
```diff
@@ -81,5 +83,13 @@ export function ResponsiveDropdownTest() {
 
       expect(handler).toHaveBeenCalled();
     });
+
+    it('should call close when the dropdown is opened and the escape key is pressed', () => {
+      responsiveDropdown.close = jasmine.createSpy('close');
+      responsiveDropdown.open();
```
Comment (samisayegh) [inline unitTests/ui/ResponsiveComponents/ResponsiveDropdownTest.ts]: Depending on the comment above, you could remove the call to `open()` and adjust the test description
```
should call close when the escape key is pressed
```

File: src/ui/ResponsiveComponents/ResponsiveDropdown/ResponsiveDropdown.ts (hunk 2 above)
Comment (ThibodeauJF) [reply src/ui/ResponsiveComponents/ResponsiveDropdown/ResponsiveDropdown.ts]: I prefer not calling it for no reason

File: src/ui/Facet/FacetSearchElement.ts [hunk 4]
```diff
@@ -235,10 +253,12 @@ export class FacetSearchElement {
   private buildInputElement() {
     return <HTMLInputElement>$$('input', {
       className: 'coveo-facet-search-input',
-      type: 'test',
+      type: 'text',
```
Comment (olamothe) [inline src/ui/Facet/FacetSearchElement.ts]: `test`.. Oh my ... 🤦‍♂️

File: strings/strings.json [hunk 5]
```diff
@@ -2674,6 +2674,14 @@
     "zh-cn": "展开",
     "zh-tw": "展開"
   },
+  "CollapseFacet": {
+    "en": "Collapse {0} facet",
```
Comment (olamothe) [inline strings/strings.json]: I'm not sure I like the term `facet`here, since this is meant to be read to end users, and not "implementer".

Maybe `Collapse filter` / `Expand filter` ...?  🤔

File: src/ui/Facet/FacetSearchElement.ts (hunk 4 above)
Comment (ThibodeauJF) [reply src/ui/Facet/FacetSearchElement.ts]: hehe

File: strings/strings.json (hunk 5 above)
Comment (ThibodeauJF) [reply strings/strings.json]: We should change it in other places then, there are a lot of strings containing facet:
ShowFacet
HideFacet
SaveFacetState
ClearFacetState
Facets
FacetTitle
I think Facet is fine when talking about the UI component
Facets apply filters ....
Or we can change the facet for filter in most these places
What do you think @fbeaudoincoveo  ?

Comment (fbeaudoincoveo) [reply strings/strings.json]: Mmm for consistency's sake, we'd probably want to update these occurrences to the equivalent word for "filter" in other languages too.

E.g., "Agrandir le filtre {0}"

Sounds pretty weird in French :)

I agree with JF: I believe using _facet_ is fine when referring to the UI component; if we're talking about selecting facet _values_ then we may want to consider using "filter" in some instances. That'd really be a case by case thing, though.

In short, I think we should leave it to "Collapse {0} facet".
